cmd/thanos/query.go

```diff
@@ -478,6 +492,25 @@ func runQuery(
 	return nil
 }
 
+func runtimeInfo(logger log.Logger) v1.RuntimeInfo {
+	cwd, err := os.Getwd()
+	if err != nil {
+		cwd = "<error retrieving current working directory>"
+		level.Warn(logger).Log("msg", "failed to retrieve current working directory", "err", err)
+	}
+
+	status := v1.RuntimeInfo{
+		StartTime:      time.Now().UTC(),
+		CWD:            cwd,
```

Comment:
Oops! There is a bug here that completely slipped past me. Every time the API gets a request for status, we run `runtimeinfo()` to get updated values for `GoroutineCount` etc. but we are also setting `StartTime` to `time.now().UTC()` So we are sending current time instead of the start time :open_mouth: 

We can make this a method of the API itself like Prometheus is doing it here https://github.com/prometheus/prometheus/blob/98a69b77d1eda71c20836cbebf83b956c68de3a9/web/web.go#L817 but then, I am not sure how can we expose this info to `NewQueryUI()` as well.

@GiedriusS @squat wdyt?

Reply:
Sorry for the late response. What about making the function `runtimeInfo` in-line that "catches" `CWD` (not necessarily but it will most likely never change, right?) and `Birth` from the environment, and then passing it to both `ui.NewQueryUI()` and `v1.NewAPI()`? Something like this: https://play.golang.org/p/KG6Rf_vns32.